### PR TITLE
Support pasting of numbers, use HTML5 number type

### PR DESCRIPTION
**Pull request by f3ndot**
paste

* No longer tries to use JS to prevent particular, non-numeric characters (HTML5 input number)
* Grabs pasting data, and fills as much of the form as possible-- clobbering existing input
* Hits the `onCompleted` callback if paste fills all inputs
* Have to use a little bit of JS to prevent numbers with more than one digit (`mix`, `max` aren't being respected)

THIS WAS HARD!

### CODE REVIEW

File: src/multiple-input/_ws-multiple-input.scss [hunk 1]
```diff
@@ -8,8 +8,18 @@
...
+  input[type=number] {
+    -moz-appearance: textfield;
+    appearance: textfield;
+
+    &::-webkit-inner-spin-button,
+    &::-webkit-outer-spin-button {
+      -webkit-appearance: none;
+      margin: 0;
+    }
+  }
```
Comment (f3ndot) [inline src/multiple-input/_ws-multiple-input.scss]: StackOverflow tells me this is the way to hide those god-awful shadowdom buttons for number input

File: src/multiple-input/ws-multiple-input.controller.js [hunk 2]
```diff
@@ -13,7 +19,7 @@ export default class WsMultipleInputController {
       // When maximum length is reached for current input, not counting movements
       if ($event.target.value.length === $event.target.maxLength && !($event.key === 'ArrowLeft' || $event.key === 'ArrowRight')) {
         const inputs = this.model.inputs;
-        const populated = inputs.filter(input => angular.isString(input) && input.length > 0);
+        const populated = inputs.filter(input => (angular.isString(input) && input.length > 0) || angular.isNumber(input));
```
Comment (cchawn) [inline src/multiple-input/ws-multiple-input.controller.js]: this line is [too long](https://circleci.com/gh/wealthsimple/ws-directives/644?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link)

File: src/multiple-input/ws-multiple-input.controller.js [hunk 3]
```diff
@@ -28,11 +34,9 @@ export default class WsMultipleInputController {
 
   onKeyDown = ($event) => {
     if (!this.isReadonly) {
-      // Because ng-pattern doesn't disallow characters from being inputted, we have to forbid
-      // alphabet keys (but allow modifiers, backspace, etc.)
-      if (this.isNumeric && RegExp('^[^0-9]$').test($event.key)) {
+      // Browsers have a hard time limiting input for <input type="number">
+      if (($event.target.value.length > 0 && $event.keyCode >= 48 && !$event.altKey && !$event.ctrlKey && !$event.metaKey)) {
```
Comment (cchawn) [inline src/multiple-input/ws-multiple-input.controller.js]: and this one 😬

File: src/multiple-input/_ws-multiple-input.scss (hunk 1 above)
Comment (cchawn) [reply src/multiple-input/_ws-multiple-input.scss]: Looks good to me! Should verify that this works for IE (using Browserstack credentials in 1Password)

File: src/multiple-input/ws-multiple-input.tmpl.html [hunk 4]
```diff
@@ -1,10 +1,12 @@
 <div class="ws-multiple-input-section">
-  <input type="text"
+  <input
+    type="{{$ctrl.isNumeric ? 'number' : 'text'}}"
```
Comment (cchawn) [inline src/multiple-input/ws-multiple-input.tmpl.html]: Should we also set a minimum value when the input is type number? (To restrict users from entering/pressing the down arrow key to get negative numbers)

**Pull request merged**